sdks/python/apache_beam/ml/inference/xgboost_inference.py

```diff
@@ -0,0 +1,212 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+import sys
+from abc import ABC
+from typing import Any
+from typing import Callable
+from typing import Dict
+from typing import Iterable
+from typing import Optional
+from typing import Sequence
+from typing import Union
+
+import datatable
+import numpy
+import pandas
+import scipy
+import xgboost
+
+from apache_beam.ml.inference.base import ExampleT
+from apache_beam.ml.inference.base import ModelHandler
+from apache_beam.ml.inference.base import ModelT
+from apache_beam.ml.inference.base import PredictionResult
+from apache_beam.ml.inference.base import PredictionT
+
+
+class XGBoostModelHandler(ModelHandler[ExampleT, PredictionT, ModelT], ABC):
+  def __init__(
+      self,
+      model_class: Union[Callable[..., xgboost.Booster],
+                         Callable[..., xgboost.XGBModel]],
+      model_state: str):
+    self.model_class = model_class
+    self.model_state = model_state
+
+  def load_model(self) -> Union[xgboost.Booster, xgboost.XGBModel]:
+    model = self.model_class()
+    model.load_model(self.model_state)
```

Comment:
Could we extend this so that it can accept arbitrary Beam Filesystem locations (e.g. gcs) so that it can easily be used on remote runners? You should be able to do that by using the Filesystem IO utilities - https://github.com/apache/beam/blob/c0e689331c2a6573ecf267b9bef133a85ea8a36c/sdks/python/apache_beam/io/filesystem.py#L810 - if there's not a good way to load the model via file stream, its ok to save the model locally before loading

Reply:
I used the Apache Beam `open` method from the FileSystem modude. After reading the file's data from the filehandler, I converted it into a Python bytestream that can be loaded by XGBoost.